sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -448,6 +448,14 @@ class Analyzer(
       }
     }
 
+    /**
+     * Data source formats that were not supported in direct query on file
+     */
+    private final val unsupportedFileQuerySource = Set(
+      "org.apache.spark.sql.jdbc",
```

Comment:
Not sure about this. Looks hard to maintain. E.g., you forget to add `JdbcRelationProvider`. @rxin What do you think?